tests/lang_objects/destructor/ttypeops_misc.nim

```diff
@@ -0,0 +1,18 @@
+discard """
+"""
+
+block ensure_lifting_is_not_confused_by_type_wrappers:
+  ## previously this led to a code gen bug where a non-existent hidden type
+  ## field was being copied. the various wrappers alias/distinct weren't being
+  ## skipped when testing to see if the underlying type was an object.
+  type
+    Object = object
+      x: string # force `Object` to require lifetime hooks
```

Comment:
```suggestion
  type WithHooks = object

  proc `=destroy`(x: var WithHooks) =
    # force `WithHooks` to require lifetime hooks
    discard

  type
    Object = object
      x: WithHooks # make sure that `Object` gets a synthesized copy hook
```
A soft suggestion for making the test-case independent of `--gc:arc|orc`, as only with the latter two does the `string` type have hooks.

Reply:
good call, independence would be much better.